perfkitbenchmarker/linux_virtual_machine.py

```diff
@@ -660,3 +678,131 @@ def GetServiceName(self, package_name):
+  Commands within a Docker Container running Ubuntu"""
+
+  def OnStartup(self):
+    """Initializes docker before proceeding with Startup."""
+    self.RemoteHostCommand('mkdir -p %s' % vm_util.VM_TMP_DIR)
+    self.Install('docker')
+    self.InitDocker()
+    super(ContainerizedDebianMixin, self).OnStartup()
+
+  def InitDocker(self):
+    """Initializes the docker container daemon."""
+    init_docker_cmd = 'sudo docker run -d --net=host '
+    for disk in self.scratch_disks:
+      init_docker_cmd += '-v %s:%s ' % (disk.mount_point, disk.mount_point)
+    init_docker_cmd += '%s sleep infinity ' % UBUNTU_DOCKER_IMAGE
+
+    resp, _ = self.RemoteHostCommand(init_docker_cmd)
+    self.docker_id = resp[:-1]
+    return self.docker_id
+
+  def RemoteCommand(self, command,
+                    should_log=False, retries=SSH_RETRIES,
+                    ignore_failure=False, login_shell=False,
+                    suppress_warning=False):
+    """Runs a command inside the container.
+
+    Args:
+      command: A valid bash command.
+      should_log: A boolean indicating whether the command result should be
+          logged at the info level. Even if it is false, the results will
+          still be logged at the debug level.
+      retries: The maximum number of times RemoteCommand should retry SSHing
+          when it receives a 255 return code.
+      ignore_failure: Ignore any failure if set to true.
+      login_shell: Run command in a login shell.
+      suppress_warning: Suppress the result logging from IssueCommand when the
+          return code is non-zero.
+
+    Returns:
+      A tuple of stdout and stderr from running the command.
+    """
+    # Escapes bash sequences
+    command = command.replace('\'', '\'\\\'\'')
+
+    logging.info("Docker running: %s" % command)
+    command = "sudo docker exec %s bash -c '%s'" % (self.docker_id, command)
+    return self.RemoteHostCommand(command, should_log, retries,
+                                  ignore_failure, login_shell, suppress_warning)
+
+  def ContainerCopy(self, host_path, container_path='', copy_to=True):
+    """Copies a file from host to container and vice versa.
+
+    Does not preserve permissions. If copying from container, host_path
+    must be a directory.
+
+    Args:
+      host_path: Path to file in the host.
+      container_path: Optional path of where to copy file on container..
+      copy_to: True to copy to container, False to copy from container.
+
+    Raises:
+      RemoteExceptionError: If the source container_path is blank.
+    """
+    if copy_to:
+      file_name = os.path.basename(host_path)
+
+      # Default container directory is /
+      if container_path == '':
+        container_path = '/%s' % file_name
+
+      # Need to ensure container_path is not a directory
+      if container_path[-1] == '/':
+        container_path = os.path.join(container_path, file_name)
+
+      command = "cat %s | sudo docker exec -i %s bash -c 'cat > %s'" \
```

Comment:
Is this the recommended way to do the copy? The solution here seems cleaner to me:
http://stackoverflow.com/questions/22907231/copying-files-from-host-to-docker-container